test/profiler/test_cpp_thread.py

```diff
@@ -85,6 +88,8 @@ def tearDownClass(cls):
     def setUp(self) -> None:
         if not torch.cuda.is_available():
             self.skipTest("Test machine does not have cuda")
+        global device
```

Comment:
What's the motivation to make `device` a global variable? Does a class member variable serve the same purpose?

Reply:
Because the global device is used by the function `emulateTraining` https://github.com/pytorch/pytorch/pull/134316/files#diff-5492234fdf4fe8192416757b9a3fe2c7e5e0d05786d1d847ddc42f2f466ac972R60-R61. This function is underlying called from a CPP file `test/profiler/test_cpp_thread.cpp`. We need to use global device to pass which device is being tested to `emulateTraining`.